osbuild/buildroot.py

```diff
@@ -232,13 +234,18 @@ def run(self, argv, monitor, binds=None, readonly_binds=None):
 
         data = io.StringIO()
 
-        while True:
-            txt = proc.stdout.read(4096)
-            if not txt:
-                break
-
-            data.write(txt)
-            monitor.log(txt)
+        with select.epoll() as epoll:
+            epoll.register(proc.stdout.fileno(), select.EPOLLIN)
+            while True:
+                events = epoll.poll(maxevents=1)
+                fileno, _ = events[0]
+                buf = os.read(fileno, 32768)
+                if not buf:
+                    break
+
+                txt = buf.decode("utf-8")
+                data.write(txt)
+                monitor.log(txt)
```

Comment:
Wouldn't it be enough to use `os.read()` in blocking mode without epoll? The `read()` syscall does not wait *IF* there is data.

Reply:
:+1: I was briefly worried that `os.read` was more "clever" than plain read, but then of course using `poll` might even help with that. After a look at the Python source, I think we are fine though. Makes the code simpler :heart: